### PR TITLE
Add support for the nRF54H20 DK board

### PR DESCRIPTION
**Pull request by anangl**
Add initial support for the nRF54H20 (not EngA) SoC and the nRF54H20 DK board.
Replace nRF54H20 PDK with DK.

### CODE REVIEW

**Comment by anangl** [comment]
Added some missing stuff, removed changes in snippets and tests.

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Removed nRF54H20 PDK.

File: boards/nordic/nrf54h20dk/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+The nRF54H20 DK is a single-board development kit for evaluation and development
+on the Nordic nRF54H20 System-on-Chip (SoC).
+
+The nRF54H20 is a multicore SoC with:
+
+* an Arm Cortex-M33 core with DSP instructions, FPU, and Armv8-M Security
+  Extensions, running at up to 320 MHz, referred to as the **application core**
+* an Arm Cortex-M33 core with DSP instructions, FPU, and Armv8-M Security
+  Extensions, running at up to 256 MHz, referred to as the **radio core**.
+
```
Comment (nordicjm) [inline boards/nordic/nrf54h20dk/doc/index.rst:23]: Shouldn't this describe ppr?

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+identifier: nrf54h20dk/nrf54h20/cpuapp
+name: nRF54H20-DK-nRF54H20-Application
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 32
+flash: 368
```
Comment (nordicjm) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml]: these numbers can't really be accurate can they?

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpurad.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+identifier: nrf54h20dk/nrf54h20/cpurad
+name: nRF54H20-DK-nRF54H20-Radio
+type: mcu
+arch: arm
+toolchain:
+  - gnuarmemb
+  - xtools
+  - zephyr
+ram: 32
+flash: 368
```
Comment (nordicjm) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpurad.yaml]: As above

File: boards/nordic/nrf54h20dk/doc/index.rst [hunk 4]
```diff
@@ -42,51 +41,53 @@ nRF54H20 SoC provides support for the following devices:
 * :abbr:`USB (Universal Serial Bus)`
 * :abbr:`WDT (Watchdog Timer)`
 
-.. figure:: img/nrf54h20pdk_nrf54h20.webp
-     :align: center
-     :alt: nRF54H20 PDK
-
-     nRF54H20 PDK (Credit: Nordic Semiconductor)
-
```
Comment (nordicjm) [inline boards/nordic/nrf54h20dk/doc/index.rst:50]: No picture?

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/nordic/common/vpr/Kconfig.defconfig [hunk 5]
```diff
@@ -5,7 +5,7 @@ CPU_PATH := $(dt_nodelabel_path,cpu)
 CPU_ID := $(dt_node_reg_addr_int,$(CPU_PATH))
 
 config RV_BOOT_HART
-	default $(CPU_ID)
+	default $(CPU_ID) if SOC_NRF54H20_ENGA_CPUPPR
```
Comment (gmarull) [inline soc/nordic/common/vpr/Kconfig.defconfig]: why is this needed?

File: boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NRF54H20DK
+	select SOC_NRF54H20_CPUAPP if BOARD_NRF54H20DK_NRF54H20_CPUAPP
+	select SOC_NRF54H20_CPURAD if BOARD_NRF54H20DK_NRF54H20_CPURAD
+	select SOC_NRF54H20_CPUPPR if BOARD_NRF54H20DK_NRF54H20_CPUPPR || \
+				      BOARD_NRF54H20DK_NRF54H20_CPUPPR_RAM
```
Comment (gmarull) [inline boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk:8]: we should now dexault to CONFIG_XIP=n for PPR, and introduce the `/xip` variant. EngA was done the other way around because 28K of RAM + Zephyr defaults lead to an unusable platform.

File: boards/nordic/nrf54h20dk/board.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+board:
+  name: nrf54h20dk
+  vendor: nordic
+  socs:
+  - name: nrf54h20
+    variants:
+    - name: ram
+      cpucluster: cpuppr
```
Comment (gmarull) [inline boards/nordic/nrf54h20dk/board.yml]: as said: change this to `xip`, also, adjust the snippets so that we have a `nordic-ppr-xip`

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 1 above)
Comment (gmarull) [reply boards/nordic/nrf54h20dk/doc/index.rst:23]: we should

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml (hunk 2 above)
Comment (gmarull) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml]: ram: 256k

File: soc/nordic/common/vpr/Kconfig.defconfig [hunk 8]
```diff
@@ -1,12 +1,6 @@
 # Copyright (c) 2024 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
-CPU_PATH := $(dt_nodelabel_path,cpu)
-CPU_ID := $(dt_node_reg_addr_int,$(CPU_PATH))
-
-config RV_BOOT_HART
-	default $(CPU_ID) if SOC_NRF54H20_ENGA_CPUPPR
-
```
Comment (gmarull) [inline soc/nordic/common/vpr/Kconfig.defconfig:9]: isn't this needed for FP1?

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml (hunk 2 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml]: Updated.

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpurad.yaml (hunk 3 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpurad.yaml]: Updated.

File: boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk (hunk 6 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk]: done

File: boards/nordic/nrf54h20dk/board.yml (hunk 7 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/board.yml]: done

File: soc/nordic/common/vpr/Kconfig.defconfig (hunk 8 above)
Comment (anangl) [reply soc/nordic/common/vpr/Kconfig.defconfig]: PPR on FP1 won't boot if this is not set to 0 (which is the default value).

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 4 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst:50]: Not yet. I'll add it when I get one.

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 1 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst:23]: Yes, info about PPR and its targets will be added.

File: include/zephyr/dt-bindings/misc/nordic-nrf-ficr-nrf54h20.h [hunk 9]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* autogenerated using Nordic HAL utils/gen_offsets.py script */
+
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_MISC_NORDIC_NRF_FICR_NRF54H20_H_
```
Comment (gmarull) [inline include/zephyr/dt-bindings/misc/nordic-nrf-ficr-nrf54h20.h:8]: shouldn't the enga header be deleted?

Comment (anangl) [reply include/zephyr/dt-bindings/misc/nordic-nrf-ficr-nrf54h20.h:8]: It is deleted in cca0d2df083f9b4b8146a5ea2cb85db9e3318bb4.

**Review by gmarull — DISMISSED** [review]

File: boards/nordic/nrf54h20dk/doc/index.rst [hunk 10]
```diff
@@ -129,18 +130,18 @@ Follow the instructions in the :ref:`nordic_segger` page to install
 and configure all the necessary software. Further information can be
 found in :ref:`nordic_segger_flashing`.
 
-To build and program the sample to the nRF54H20 PDK, complete the following steps:
+To build and program the sample to the nRF54H20 DK, complete the following steps:
 
-First, connect the nRF54H20 PDK to you computer using the IMCU USB port on the PDK.
+First, connect the nRF54H20 DK to you computer using the IMCU USB port on the DK.
```
Comment (FrancescoSer) [inline boards/nordic/nrf54h20dk/doc/index.rst:135]:
```suggestion
1. Connect the nRF54H20 DK to your computer using the IMCU USB port on the DK.
```

File: boards/nordic/nrf54h20dk/doc/index.rst [hunk 11]
```diff
@@ -129,18 +130,18 @@ Follow the instructions in the :ref:`nordic_segger` page to install
...
 Next, build the sample by running the following command:
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
```
Comment (FrancescoSer) [inline boards/nordic/nrf54h20dk/doc/index.rst:139]:
```suggestion
#. Build the sample by running the following command:

   .. zephyr-app-commands::
      :zephyr-app: samples/hello_world
```

File: boards/nordic/nrf54h20dk/doc/index.rst [hunk 12]
```diff
@@ -129,18 +130,18 @@ Follow the instructions in the :ref:`nordic_segger` page to install
...
-   :board: nrf54h20pdk/nrf54h20/cpuapp
+   :board: nrf54h20dk/nrf54h20/cpuapp
    :goals: build flash
```
Comment (FrancescoSer) [inline boards/nordic/nrf54h20dk/doc/index.rst:141]:
```suggestion
      :board: nrf54h20dk/nrf54h20/cpuapp
      :goals: build flash
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Just a pair of suggestions

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 10 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst]: Applied.

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 11 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst]: Applied.

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 12 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst]: Applied.

**Comment by anangl** [comment]
Rebased.

**Comment by anangl** [comment]
Rebased.

File: boards/nordic/nrf54h20dk/doc/index.rst (hunk 1 above)
Comment (anangl) [reply boards/nordic/nrf54h20dk/doc/index.rst:23]: @nordicjm How about adding it in a follow-up PR together with the picture that is not available yet?

Comment (nordicjm) [reply boards/nordic/nrf54h20dk/doc/index.rst:23]: > @nordicjm How about adding it in a follow-up PR together with the picture that is not available yet?

Works for me

**Review by gmarull — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by FrancescoSer — APPROVED** [review]
